File: problem010.py
import math

limit = 2000000

not_prime = []

def is_prime(n):
    if n in not_prime:
        return False

    i = 2
    while i < n:
        if (n % i != 0):
            not_prime.append(n)
            return False
        i += 1
    return True

def main():
    i = 2
    total = 0

    while i < limit:
        print i
        if is_prime(i):
            total += i

        i += 1

    print total

if __name__ == '__main__':
  main()

File: problem010.rs
use std::num::sqrt;

fn is_prime(n: int) -> bool {
    let upper_limit = sqrt(n as f64) + 1.0;
    for i in range(2, upper_limit as int) {
        if n % i == 0 {
            return false;
        }
    }
    return true;
}

fn main() {
    let mut primes = ~[2, 3, 5, 7];

    let mut i = 8;
    while i < 2000000 {
        if is_prime(i) {
            primes.push(i);
        }
        i = i + 1;
    }

    let mut total = 0;
    for &x in primes.iter() {
        total += x
    }

    println!("{}", total);
}
